packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2557,31 +2595,40 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
         child._needsSemanticsGeometryUpdate = true;
       }
       final _SemanticsFragment fragment = child._getSemanticsFragment();
-      if (fragment != null) {
+      assert(fragment != null);
+      if (fragment.dropSemanticsOfPreviousSiblings) {
+        children = null; // throw away all left siblings of [child].
+        dropSemanticsOfPreviousSiblings = true;
+      }
+      if (fragment.producesSemanticNodes) {
         fragment.addAncestor(this);
         children ??= <_SemanticsFragment>[];
         assert(!children.contains(fragment));
         children.add(fragment);
       }
     });
+    if (isSemanticBoundary) {
+      // Don't propagate [dropSemanticsOfPreviousSiblings] past a semantic boundary.
+      dropSemanticsOfPreviousSiblings = false;
```

Comment:
Oh this is about propagating _up_. I somehow was confused and thought it was propagating down.